devel/cutter-r2/Portfile

```diff
@@ -0,0 +1,73 @@
+PortGroup           github 1.0
+PortGroup           qmake5 1.0
+
+github.setup        radareorg cutter 1.10.3 v
+name                cutter-r2
+revision            0
+
+categories          devel
+platforms           darwin
+license             GPL-3
+maintainers         {@i0ntempest me.com:szf1234} openmaintainer
+
+description         Free and Open Source RE Platform powered by radare2
+long_description    Cutter is a free and open-source reverse engineering framework\
+                    powered by radare2. Its goal is making an advanced, customizable\
+                    and FOSS reverse-engineering platform while keeping the user\
+                    experience at mind. Cutter is created by reverse engineers for\
+                    reverse engineers.
+
+homepage            https://cutter.re/
+
+checksums           rmd160  e2c88f2eb32c0ed6cee9e8cd3bc71ddf69363ed3 \
+                    sha256  4a65e5c2a02e0dd7d7359ede2f0ebb5c9d0e3411d6718f6aad57ba9affe53717 \
+                    size    1866181
+
+depends_lib-append  port:radare2
+qt5.depends_component \
+                    qtsvg
+                    
+compiler.cxx_standard 2011
+configure.args-append \
+                    src/Cutter.pro \
+                    INCLUDEPATH+=${prefix}/include/libr
+
+destroot {
+        copy ${worksrcpath}/Cutter.app ${destroot}${applications_dir}
+}
+
+variant python38 description {Enable Python support and bindings using Python 3.8} {
+        depends_lib-append  port:python38 \
+                            port:py38-pyside2
+
+        set python_framework \
+                            ${prefix}/Library/Frameworks/Python.framework
+
+        configure.args-append \
+                            CONFIG+=CUTTER_ENABLE_PYTHON \
+                            CONFIG+=CUTTER_ENABLE_PYTHON_BINDINGS \
+                            INCLUDEPATH+=${python_framework}/Versions/3.8/include/python3.8/ \
+                            LIBS+="-L${python_framework}/Versions/3.8/lib -lpython3.8" \
+                            QMAKE_RPATHDIR+=${python_framework}/Versions/3.8/lib/python3.8/site-packages/shiboken2/ \
+                            QMAKE_RPATHDIR+=${python_framework}/Versions/3.8/lib/python3.8/site-packages/PySide2/ \
+                            PYTHON_FRAMEWORK_DIR=${python_framework} \
+                            SHIBOKEN_EXECUTABLE=${python_framework}/Versions/3.8/bin/shiboken2 \
+                            SHIBOKEN_INCLUDEDIR=${python_framework}/Versions/3.8/lib/python3.8/site-packages/shiboken2_generator/include/ \
+                            SHIBOKEN_LIBRARY=${python_framework}/Versions/3.8/lib/python3.8/site-packages/shiboken2/libshiboken2.cpython-38-darwin.5.15.dylib \
+                            PYSIDE_INCLUDEDIR=${python_framework}/Versions/3.8/lib/python3.8/site-packages/PySide2/include/ \
+                            PYSIDE_LIBRARY=${python_framework}/Versions/3.8/lib/python3.8/site-packages/PySide2/libpyside2.cpython-38-darwin.5.15.dylib \
```

Comment:
The hardcoded "5.15" for the pyside2 and shiboken libraries means this port will break as soon as those library versions change. That means two things:

1. Whenever pyside2 is updated and its library version changes, this port will need to be revbumped. A comment should be added to the top of the pyside2 port so that whoever does that update will know to do that.
2. The "5.15" in this port will need to be changed to whatever the new library version is. There are 2 ways to avoid this:
   1. pyside2 could be modified to install a symlink without the version number; you could then use the symlink here.
   2. You could inspect the filesystem and use whichever version of the library is actually there. This will require delaying appending those configure.args until pre-configure time, because before that time the files might not be there. So for example:

```
pre-configure {
    configure.args-append \
                            SHIBOKEN_LIBRARY=[glob ${python_framework}/Versions/3.8/lib/python3.8/site-packages/shiboken2/libshiboken2.cpython-38-darwin.*.dylib] \
                            PYSIDE_LIBRARY=[glob ${python_framework}/Versions/3.8/lib/python3.8/site-packages/PySide2/libpyside2.cpython-38-darwin.*.dylib]
}
```